swarm-ui/src/routes/(app)/connect/+page.svelte

```diff
@@ -61,33 +59,32 @@
 			return
 		}
 
-		// Get parameters from URL
-		const origin = $page.url.searchParams.get('origin')
-		if (!origin) {
-			error = 'No origin parameter found in URL'
-			return
-		}
+		if (!sessionStore.data.appOrigin) {
+			// Get parameters from URL
+			const appOrigin = page.url.searchParams.get('origin')
```

Comment:
I wonder if we actually need this as a search param and not just hash param since this is single page app. Clearing search parameters causes re-render unlike hash params.